packages/core/src/transports/batchExecutor.ts

```diff
@@ -0,0 +1,89 @@
+import { murmur3 } from 'murmurhash-js';
+
+import type { TransportItem } from '../';
+
+import type { BatchExecutorOptions, SendFn } from './types';
+
+const DEFAULT_BATCH_SEND_TIMEOUT_MS = 250;
+const DEFAULT_BATCH_SEND_COUNT = 50;
+
+export class BatchExecutor {
+  private readonly batchSendCount: number;
+  private readonly batchSendTimeout: number;
+
+  private signalBuffer: TransportItem[] = [];
+  private sendFn: SendFn;
+  private paused: boolean;
+  private flushInterval: number;
+
+  constructor(sendFn: SendFn, options?: BatchExecutorOptions) {
+    this.batchSendCount = options?.batchSendCount ?? DEFAULT_BATCH_SEND_COUNT;
+    this.batchSendTimeout = options?.batchSendTimeout ?? DEFAULT_BATCH_SEND_TIMEOUT_MS;
+    this.paused = options?.paused || false;
+    this.sendFn = sendFn;
+    this.flushInterval = -1;
+
+    if (options?.autoStart) {
```

Comment:
What user goal do you have in mind by providing a austoStart option?
Wouln't it be enough to say bacthing on or off?

Reply:
Ah or is this the replacement for `batchEnabled` which is documented in the [README](https://github.com/grafana/faro-web-sdk/blob/9ab0d4d5433e0cd90a20714417775e74d0dc2638/packages/core/README.md#L39) ?